### PR TITLE
Bucket Lifecycle configuration APIs(Get, Set, Remove)

### PR DESCRIPTION
**Pull request by BigUstad**
Args, Examples & Functional test

### CODE REVIEW

File: Minio.Examples/Cases/GetBucketLifecycle.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/*
+ * MinIO .NET Library for Amazon S3 Compatible Cloud Storage, (C) 2020 MinIO, Inc.
```
Comment (kannappanr) [inline Minio.Examples/Cases/GetBucketLifecycle.cs:2]:
```suggestion
 * MinIO .NET Library for Amazon S3 Compatible Cloud Storage, (C) 2021 MinIO, Inc.
```

File: Minio/DataModel/AbortIncompleteMultipartUpload.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
+/*
```
Comment (kannappanr) [inline Minio/DataModel/AbortIncompleteMultipartUpload.cs:1]: shouldn't this be under `ilm` folder?

File: Minio/DataModel/BucketOperationsArgs.cs [hunk 3]
```diff
@@ -430,6 +430,9 @@ public override RestRequest BuildRequest(RestRequest request)
             request.AddQueryParameter("tagging","");
             string body = this.BucketTags.MarshalXML();
             request.AddParameter(new Parameter("text/xml", body, ParameterType.RequestBody));
+            request.AddOrUpdateParameter("Content-MD5",
```
Comment (kannappanr) [inline Minio/DataModel/BucketOperationsArgs.cs:433]: is this computed for all operations?

Comment (BigUstad) [reply Minio/DataModel/BucketOperationsArgs.cs:433]: All operations with XML configuration attached, we get an exception if 'Content-MD5' is missing

**Review by krisis — APPROVED** [review]

**Review by kannappanr — APPROVED** [review]
LGTM

**Pull request merged**